src/overlays/actors/ovl_En_GeldB/z_en_geldb.c

```diff
@@ -33,93 +97,1466 @@ const ActorInit En_GeldB_InitVars = {
+    static Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A38348.s")
+    Vec3f swordTip;
+    Vec3f swordHilt;
+    EnGeldB* this = THIS;
+    s32 effectIndex = -1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A38430.s")
+    if (limbIndex == 11) {
+        Matrix_MultVec3f(&swordQuadOffset1, &this->swordCollider.dim.quad[1]);
+        Matrix_MultVec3f(&swordQuadOffset0, &this->swordCollider.dim.quad[0]);
+        Matrix_MultVec3f(&swordQuadOffset3, &this->swordCollider.dim.quad[3]);
+        Matrix_MultVec3f(&swordQuadOffset2, &this->swordCollider.dim.quad[2]);
+        func_80062734(&this->swordCollider, &this->swordCollider.dim.quad[0], &this->swordCollider.dim.quad[1],
+                      &this->swordCollider.dim.quad[2], &this->swordCollider.dim.quad[3]);
+        Matrix_MultVec3f(&swordTipOffset, &swordTip);
+        Matrix_MultVec3f(&swordHiltOffset, &swordHilt);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A384E8.s")
+        if ((this->swordState < 0) ||
+            ((this->actionState != GELDB_SLASH) && (this->actionState != GELDB_SPIN_ATTACK))) {
+            EffectBlure_AddSpace(Effect_GetByIndex(this->blureIdx));
+            this->swordState = 0;
+        } else if (this->swordState > 0) {
+            EffectBlure_AddVertex(Effect_GetByIndex(this->blureIdx), &swordTip, &swordHilt);
+        }
+    } else {
+        func_8002BDB0(&this->actor, limbIndex, 19, &footOffset, 22, &footOffset);
+    }
+    if (limbIndex == 19) {
+        Matrix_MultVec3f(&footOffset, &this->rightFootPos);
+    } else if (limbIndex == 22) {
+        Matrix_MultVec3f(&footOffset, &this->leftFootPos);
+    }
+    if (this->iceTimer != 0) {
+        switch (limbIndex) {
+            case 3:
+                effectIndex = 0;
+                break;
+            case 16:
+                effectIndex = 1;
+                break;
+            case 11:
+                effectIndex = 2;
+                break;
+            case 12:
+                effectIndex = 3;
+                break;
+            case 7:
+                effectIndex = 4;
+                break;
+            case 2:
+                effectIndex = 5;
+                break;
+            case 23:
+                effectIndex = 6;
+                break;
+            case 19:
+                effectIndex = 7;
+                break;
+            case 22:
+                effectIndex = 8;
+                break;
+            default:
+                break;
+        }
+        if (effectIndex >= 0) {
+            Vec3f limbPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A387D0.s")
+            Matrix_MultVec3f(&zeroVec, &limbPos);
+            this->effectPos[effectIndex].x = limbPos.x;
+            this->effectPos[effectIndex].y = limbPos.y;
+            this->effectPos[effectIndex].z = limbPos.z;
```

Comment:
this is more accurately described as `bodyPartsPos[]` and the index being `bodyPart`
the effect will then use the array of body parts positions to place the effect